controllers/kubeletconfig_controller.go

```diff
@@ -140,6 +144,15 @@ func (r *KubeletConfigReconciler) SetupWithManager(mgr ctrl.Manager, clusterPlat
 			_, ok := kubelet.Labels[HypershiftKubeletConfigConfigMapLabel]
 			return ok
 		})
+		p.DeleteFunc = func(e event.DeleteEvent) bool {
+			kubelet := e.Object.(*corev1.ConfigMap)
+			if _, ok := kubelet.Labels[HypershiftKubeletConfigConfigMapLabel]; !ok {
+				return false
+			}
+			klog.InfoS("KubeletConfig ConfigMap object got deleted", "KubeletConfig", kubelet.Name)
+			r.GarbageCollectionFor = append(r.GarbageCollectionFor, *kubelet)
```

Comment:
maybe a deepcopy?

Reply:
`DeepCopy` still return a pointer I can make this a slice of pointers and save some redundant copying